packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -528,7 +566,8 @@ export class Carousel
   };
 
   private tabListKeyDownHandler = (event: KeyboardEvent): void => {
-    const interactiveItems = Array(...this.tabList.querySelectorAll("button"));
+    const targetEls = `button:not(.${CSS.paginationItemOutOfRange})`;
+    const interactiveItems = Array(...this.tabList.querySelectorAll(targetEls));
```

Comment:
Yeah definitely. Before this PR it could have been done just once too on slotchange.

I can still store it outside of this function, but it will still need to update more frequently than onslotchange. Setting up an "isOverflowing" state might be helpful to reduce the number of times this is set as it will not always change.